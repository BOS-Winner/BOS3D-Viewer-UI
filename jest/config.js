module.exports = {
  rootDir: '../',
  testMatch: [
    "**/__tests__/**/*.test.{js,jsx}",
  ],
  moduleNameMapper: {
    "\\.less$": "identity-obj-proxy",
    "\\.css$": "<rootDir>/jest/__mocks__/styleMock.js",
    "Libs(.*)$": "<rootDir>/src/Libs$1",
    "UIutils(.*)$": "<rootDir>/src/UI/utils$1",
    "^Base(.*)$": "<rootDir>/src/UI/Base$1",
    "^BOS3D(.*)$": "<rootDir>/jest/__mocks__/BOS3D$1",
    "^BOS2D$": "<rootDir>/jest/__mocks__/BOS2D.js",
    "^BOS3DUI$": "<rootDir>/jest/__mocks__/BOS3DUI.js",
    "^jest/(.*)$": "<rootDir>/jest/$1",
    "^mock/(.*)$": "<rootDir>/jest/__mocks__/$1",
  },
  moduleFileExtensions: ["js", "jsx"],
  moduleDirectories: ["node_modules"],
  transform: {
    "\\.(js|jsx)": "babel-jest",
    "\\.(jpg|jpeg|png|gif|svg)$": "<rootDir>/jest/__mocks__/fileMock.js",
  },
  transformIgnorePatterns: [
    // Change MODULE_NAME_HERE to your module that isn't being compiled
    // "/node_modules/(?!three|lodash-es).+\\.js$"
    "/node_modules/core-js",
    "/node_modules/@testing-library",
    "/node_modules/@babel",
  ],
  coveragePathIgnorePatterns: [
    "/node_modules/",
    "/jest/"
  ],
};
