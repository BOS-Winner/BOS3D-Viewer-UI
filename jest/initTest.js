import EventEmitter from "events";
import { createStore } from "redux";
import BOS3D from 'BOS3D';
import reducer from "../src/UI/reducer";

export function genStore() {
  const ee = new EventEmitter();
  const viewer3D = new BOS3D.Viewer();

  return createStore(reducer, {
    system: {
      viewer3D,
      BIMWINNER: { BOS3D },
      apiVersion: 'api',
      eventEmitter: ee,
    }
  });
}
