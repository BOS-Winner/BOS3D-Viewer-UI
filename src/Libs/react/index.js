import React from "react";
import useUpdateEffect from "./useUpdateEffect";

React.useUpdateEffect = useUpdateEffect;

export default React;
// export { default as useUpdateEffect } from "./useUpdateEffect";
