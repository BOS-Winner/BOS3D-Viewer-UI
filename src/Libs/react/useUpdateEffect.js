import { useEffect, useRef } from "react";

export default function (effect, deps) {
  const isFirstRender = useRef(true);
  useEffect(() => {
    if (isFirstRender.current) {
      isFirstRender.current = false;
    } else {
      effect();
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, deps);
}
