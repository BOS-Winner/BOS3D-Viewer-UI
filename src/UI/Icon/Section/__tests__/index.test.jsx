/* eslint-disable camelcase */
/* eslint-disable react/prop-types */
import React from 'react';
import { Provider } from 'react-redux';
import { genStore } from 'jest/initTest';
import { render, act } from "@testing-library/react";
import Index from '../index';

jest.mock('Base/Modal', () => props => (
  <div
    className={props.className}
    style={props.style}
  >
    {props.children}
  </div>
));
const store = genStore();
let g_getByTitle;
let indexContainer;

describe('Section index', () => {
  beforeEach(() => {
    const { container, getByTitle } = render((
      <Provider store={store}>
        <Index />
      </Provider>
    ));
    indexContainer = container;
    g_getByTitle = getByTitle;
  });

  afterAll(() => {
    jest.restoreAllMocks();
  });

  it('should render dom', () => {
    expect(indexContainer).toMatchSnapshot();
  });

  it('should trigger popup', () => {
    const popup = indexContainer.querySelector('.popup');
    const indexBtn = g_getByTitle('剖切');

    act(() => {
      indexBtn.click();
    });
    expect(popup.style.display).not.toEqual('none');
    expect(indexBtn.children[0].getAttribute('data-selected')).toEqual('false');
    act(() => {
      g_getByTitle('模型剖切').click();
    });
    expect(popup.style.display).toEqual('none');
    expect(indexBtn.children[0].getAttribute('data-selected')).toEqual('true');
    act(() => {
      indexBtn.click();
    });
    expect(popup.style.display).toEqual('none');
    expect(indexBtn.children[0].getAttribute('data-selected')).toEqual('false');

    act(() => {
      indexBtn.click();
    });
    expect(popup.style.display).not.toEqual('none');
    expect(indexBtn.children[0].getAttribute('data-selected')).toEqual('false');
    act(() => {
      g_getByTitle('自由剖切').click();
    });
    expect(popup.style.display).toEqual('none');
    expect(indexBtn.children[0].getAttribute('data-selected')).toEqual('true');
    act(() => {
      indexBtn.click();
    });
    expect(popup.style.display).toEqual('none');
    expect(indexBtn.children[0].getAttribute('data-selected')).toEqual('false');

    act(() => {
      indexBtn.click();
    });
    expect(popup.style.display).not.toEqual('none');
    expect(indexBtn.children[0].getAttribute('data-selected')).toEqual('false');
    act(() => {
      indexBtn.click();
    });
    expect(popup.style.display).toEqual('none');
    expect(indexBtn.children[0].getAttribute('data-selected')).toEqual('false');
  });
});
