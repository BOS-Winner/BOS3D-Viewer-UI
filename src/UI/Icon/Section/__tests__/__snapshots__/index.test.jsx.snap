// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section index should render dom 1`] = `
<div>
  <div>
    <div
      style="position: relative;"
    >
      <div
        class="toolbar popup"
        style="display: none;"
      >
        <div
          class=""
          role="button"
          tabindex="0"
          title="模型剖切"
        >
          <div
            class="icon "
            data-selected="false"
            role="button"
            style="background-image: url(modelSection.png);"
            tabindex="0"
          />
          <div>
            <div
              class="switch"
            >
              <div
                class="reset"
                role="presentation"
                style="background-image: url(hide@2x.png);"
                title="隐藏剖切盒"
              />
              <div
                class="eye"
                role="presentation"
                style="background-image: url(reset2.png);"
                title="重置剖切状态"
              />
            </div>
          </div>
        </div>
        <div
          class=""
          role="button"
          tabindex="0"
          title="自由剖切"
        >
          <div
            class="icon "
            data-selected="false"
            role="button"
            style="background-image: url(freeSection.png);"
            tabindex="0"
          />
          <div>
            <div
              class="freeSection"
            >
              <select
                title="改变轴向"
              >
                <option
                  selected=""
                  value="X"
                >
                  X
                </option>
                <option
                  value="Y"
                >
                  Y
                </option>
                <option
                  value="Z"
                >
                  Z
                </option>
                <option
                  value="-X"
                >
                  -X
                </option>
                <option
                  value="-Y"
                >
                  -Y
                </option>
                <option
                  value="-Z"
                >
                  -Z
                </option>
              </select>
              <img
                alt="点击隐藏剖切面"
                class="eye"
                src="show@2x.png"
                title="点击隐藏剖切面"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      role="button"
      tabindex="0"
      title="剖切"
    >
      <div
        class="icon "
        data-selected="false"
        role="button"
        style="background-image: url(section.png);"
        tabindex="0"
      />
    </div>
  </div>
</div>
`;
